BUILD.bazel

```diff
@@ -0,0 +1,46 @@
+package(default_visibility = ["//visibility:private"])
+
+genrule(
+    name = "re2c_config",
+    srcs = [
+        "CMakeLists.txt",
+        "//bazel:re2c_config.h",
+    ],
+    outs = ["config.h"],
+    cmd = """
+        cat $(location //:CMakeLists.txt) |\\
+            sed -rne 's/project\\(re2c VERSION ([^ ]+).*/#define PACKAGE_VERSION "\\1" RE2C_VERSION_SUFFIX/p' >> "$@"
```

Comment:
Thanks for simplifying. Is RE2C_VERSION_SUFFIX still needed after simplification? Here and on line 38. My understanding was that you needed it to differentiate between `re2c_bootstrap_cc` and the removed library.

Reply:
The other aspect is of course that this way I can add 'arbitrary' version suffixes to indicate that this is not a fully correct built version.

Reply:
> The overall better solution is to have a function that returns the version and have that function be implemented in either main.cc or some new verson.cc/.h file(s). In that case I can remove this here and figure out later what to do with the version. Thoughts?

A function in a separate file sounds good, provided that it will use PACKAGE_VERSION define. There's already a similar one: https://github.com/skvadrik/re2c/blob/master/src/msg/msg.cc#L99-L106, so you can add one with a suitable interface in a separate file and call it from the existing one to avoid duplicating the logic. It will require changes in source code and other build systems to add the new file, so it's better to have as a separate commit/PR.

Then for this PR you can remove `RE2C_VERSION_SUFFIX` logic. Also please rebase and squash bazel-related commits into one, then it'll be good to go.